lib/elixir/lib/macro.ex

```diff
@@ -841,7 +841,7 @@ defmodule Macro do
   defp call_to_string({:., _, [arg]}, fun),
     do: module_to_string(arg, fun) <> "."
   defp call_to_string({:., _, [left, right]}, fun),
-    do: module_to_string(left, fun) <> "." <> call_to_string(right, fun)
+    do: module_to_string(left, fun) <> "." <> if(is_atom(right), do: call_to_string_for_atom(right), else: call_to_string(right, fun))
```

Comment:
could this `is_atom` check be done in the function head and split those two cases into two heads? I think that would be cleaner than this nested if.